### ISSUE TITLE
Suggested TB Version detection patch, fixes Mac Mini 2012

### ISSUE THREAD

**Issue by owenrw** [mayankk2308/purge-wrangler#5]
I have a suggested patch to make detection of Thunderbolt version more reliable, which will ensure that the correct patch is applied on a Mac Mini 2012.

I checked this on my MM 2012 and on a MBP 2016, and the patch is applied successfully on the former, and not applied on the latter because it isn't required.

I'm guessing at the value of Thunderbolt 1, because my only TB1 device is a MM 2012 which reports itself as TB2...

Anyway, this might be useful (Was going to post in egpu.io forums, but I can't register because I'm not getting the email)

```
check_sys_iotbswitchtype()
{
  tb="$(ioreg | grep AppleThunderboltNHIType)"
  if [[ "$tb[@]" =~ "NHIType3" ]]
  then
    echo "This mac does not require the patch.\n"
    exit
  elif [[ "$tb[@]" =~ "NHIType2" ]]
  then
    sys_iotbswitchtype="$iotbswitchtype_ref"2
  elif [[ "$tb[@]" =~ "NHIType1" ]]
  then
    sys_iotbswitchtype="$iotbswitchtype_ref"1
  else
    echo "Unsupported/Invalid version of thunderbolt.\n"
    exit
  fi
}
```

**Comment by mayankk2308** [comment]
Thank you @owenrw, I shall incorporate it into the next release after testing, and then close this issue. Much appreciated.

**Comment by mayankk2308** [comment]
Resolved in **v2.0.2.**
